stack.py

```diff
@@ -88,16 +93,19 @@ def eval_program(program):
                 a = stack.pop()
             # b is the top of the stack, and a is the item before it, so
             # `... ; push 5 ; div` is dividing the result of `...` by 5.
-            if instr.op == 'add':
-                c = a + b
-            elif instr.op == 'sub':
-                c = a - b
-            elif instr.op == 'mul':
-                c = a * b
-            elif instr.op == 'div':
-                c = a / b
-            elif instr.op == 'pow':
-                c = a ** b
+            binary_ops = {
+                'add': lambda a, b: b + a,
+                'sub': lambda a, b: b - a,
+                'mul': lambda a, b: b * a,
+                'div': lambda a, b: b / a,
+                'pow': lambda a, b: b ** a,
+                'eq': lambda a, b: int(b == a),
+                'gt': lambda a, b: int(b > a),
+                'ge': lambda a, b: int(b >= a),
+                'lt': lambda a, b: int(b < a),
+                'le': lambda a, b: int(b <= a),
```

Comment:
Should all of these `int()` be `float()`? What about `← 1; ← 1; =; dup; # +; ↔; /`? With `int()` I think that'd probably give `[1, 0]`, instead of the expected `[1.0, 0.5]`.